relayer/chains/cosmos/provider.go

```diff
@@ -225,11 +225,17 @@ func (cc *CosmosProvider) AccountFromKeyOrAddress(keyOrAddress string) (out sdk.
 	return
 }
 
-func (cc *CosmosProvider) TrustingPeriod(ctx context.Context) (time.Duration, error) {
+func (cc *CosmosProvider) TrustingPeriod(ctx context.Context, overrideUnbondingPeriod time.Duration) (time.Duration, error) {
 
-	unbondingTime, err := cc.QueryUnbondingPeriod(ctx)
-	if err != nil {
-		return 0, err
+	unbondingTime := overrideUnbondingPeriod
+	var err error
+	if unbondingTime == 0 {
+		unbondingTime, err = cc.QueryUnbondingPeriod(ctx)
+		if err != nil {
+			return 0, err
+		}
+	} else {
+		unbondingTime = overrideUnbondingPeriod
```

Comment:
this else branch is unnecessary since you are already setting `unbondingTime := overrideUnbondingPeriod` at the top of the method.

Reply:
good eye!